### ISSUE TITLE
nifi-registry service never respond if via k8s port-forwarding

### ISSUE THREAD

**Issue by chwongeric** [cetic/helm-nifi#182]
Dear  cetic/helm-nifi support team,

Wish to seek help to overcome this strange issue accessing nifi-registry service via k8s port-forward.  I had k8s cluster at remote so access all services UI via k8s port-forward.   However, for accessing nifi-registry via k8s port-forward, no matter what local port I use, nifi-registry service seems never respond to request, but nifi service always works.

What is the difference between access nifi service vs access nifi-registry service via k8s port-forward?  Why nifi service always works while nifi-registry service never regardless local port chosen?  Is there any specifically need to setup for nifi-registry in this helm-chart?   Anything I might missed in value.yaml may caused this?

Here is my helm chart setup:
  nifi:
    helm:
      # https://github.com/cetic/helm-nifi/tree/0.7.8
      version: 0.7.8
    image:
      # https://hub.docker.com/r/apache/nifi/tags
      tag: 1.13.0

value.yaml contains following to enable nifi-registry pod and service

```
registry:
  enabled: true
  port: 80
  url: "nifi-registry"
```

```
% kubectl get svc  -n data-pipeline -o wide
...
nifi                                 ClusterIP   100.70.32.121    <none>        8080/TCP,9929/TCP                                                                     25h    app=nifi,release=nifi
nifi-headless                        ClusterIP   None             <none>        8080/TCP,6007/TCP                                                                     25h    app=nifi,release=nifi
nifi-registry                        ClusterIP   100.71.235.65    <none>        80/TCP                                                                                25h    app.kubernetes.io/instance=nifi,app.kubernetes.io/name=registry

```
nifi-registry service (tcp port 80) works fine if access inside k8s cluster.  However, if I try to access after k8s port-forward, it never respond.

-- from terminal A:
```
%  kubectl port-forward sts/nifi-registry --namespace data-pipeline 20337:80
Forwarding from 127.0.0.1:20337 -> 80
Forwarding from [::1]:20337 -> 80
Handling connection for 20337
E1025 11:56:34.256702   10540 portforward.go:400] an error occurred forwarding 20337 -> 80: error forwarding port 80 to pod 559b96d74c2830f98345b7a4ea031153c5dc750175197e1e02d0f04626c57e43, uid : exit status 1: 2021/10/25 18:56:34 socat[2684522] E connect(5, AF=2 127.0.0.1:80, 16): Connection refused
E1025 11:56:34.256719   10540 portforward.go:400] an error occurred forwarding 20337 -> 80: error forwarding port 80 to pod 559b96d74c2830f98345b7a4ea031153c5dc750175197e1e02d0f04626c57e43, uid : exit status 1: 2021/10/25 18:56:34 socat[2684523] E connect(5, AF=2 127.0.0.1:80, 16): Connection refused
Handling connection for 20337
```

-- from terminal B:
```
% curl -vv http://localhost:20337/nifi-registry/
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 20337 (#0)
> GET /nifi-registry/ HTTP/1.1
> Host: localhost:20337
> User-Agent: curl/7.64.1
> Accept: */*
> 
* Empty reply from server

```

However, if I port-forward to nifi (similarly to all other k8s service) via port-forward, it will respond request and access UI always works.   

```
% curl -vv http://localhost:20337/nifi          
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 20337 (#0)
> GET /nifi HTTP/1.1
> Host: localhost:20337
> User-Agent: curl/7.64.1
> Accept: */*
> 
< HTTP/1.1 302 Found
< Date: Mon, 25 Oct 2021 19:20:55 GMT
< Location: http://localhost:20337/nifi/
< Content-Length: 0
< Server: Jetty(9.4.35.v20201120)
```
 
Thank you very much for your kind help!

Regards,
Eric

**Comment by banzo** [comment]
Hello @chwongeric, thank you for the detailed explanation. The nifi-registry chart is maintained at https://github.com/dysnix/charts, you might have more success asking there.

**Comment by chwongeric** [comment]
@banzo thank you very much!  yes, this is very strange issue only via k8s port forward to remote cluster, wish me better luck for resolution there.

**Comment by chwongeric** [comment]
@banzo  I just saw this:  `[cetic/nifi] kubectl port-forward: connection refused #46`
Looks like its due to Nifi listens on the PodIP, but port-forward talks to 127.0.0.1, and fixed in Chart v0.4.2 (according to @alexnuttinck ).   I think I am hitting the same in nifi-registry.
Could you please also apply similar fix for nifi-registry here?  How do I see the fix of #46?   thanks a lot!

**Comment by mnp** [comment]
Probably not stale. Still seeing in latest.
